src/app/src/main/java/com/cmput301f21t09/budgetprojectname/views/activities/AnotherUserProfileActivity.java

```diff
@@ -0,0 +1,235 @@
+            if (retrievedCurrentUser.getSocial().containsKey(anotherUserID)) {
+                // Retrieve the value of user on screen in the logged in user's Social map
+                int valueOfAnotherUserInCurrentUserSocialMap = ((Number) retrievedCurrentUser.getSocial().get(anotherUserID)).intValue();
+
+                if (valueOfAnotherUserInCurrentUserSocialMap == 1 || valueOfAnotherUserInCurrentUserSocialMap == 2) {
+                    // Case: The logged in user is following the user on screen
+
+                    // Set to the "Following" scenario with a list of this user's public habits
+                    changeDisplayToFollowingState();
+                } else {
+                    // Case: The logged in user is not currently following the user on screen
+                    // Check if the logged in user has sent a follow request to the user on screen
+                    userController.readUser(anotherUserID, retrievedAnotherUser -> {
+                        if (retrievedAnotherUser.getSocial().containsKey(currentUserId)) {
+                            // Retrieve the value of the logged in user in user on screen's Social map
+                            int valueOfCurrentUserInAnotherUserSocialMap = ((Number) retrievedAnotherUser.getSocial().get(currentUserId)).intValue();
+
+                            if (valueOfCurrentUserInAnotherUserSocialMap == 0 || valueOfCurrentUserInAnotherUserSocialMap == 2) {
+                                // The logged in user has sent a follow request to this user
+                                changeDisplayToRequestedState();
+                            } else if (valueOfCurrentUserInAnotherUserSocialMap == 1) {
+                                // User on screen is following the logged in user but the logged in
+                                // user hasn't sent the user on screen a follow request
+
+                                followBtn.setOnClickListener(new View.OnClickListener() {
+                                    public void onClick(View v) {
+                                        requestToFollow(currentUserId, anotherUserID, 2);
+                                    }
+                                });
+                            }
+                        } else {
+                            // Case: The logged in user is not following the user on screen and has not sent the
+                            // user on screen a follow request. The user on screen is not following the logged in
+                            // user as well.
+                            followBtn.setOnClickListener(new View.OnClickListener() {
+                                public void onClick(View v) {
+                                    requestToFollow(currentUserId, anotherUserID, 0);
+                                }
+                            });
+                        }
+                    });
+                }
+            } else {
```

Comment:
I was a bit confused by the logic here. Maybe add a comment in this else case which says "we are checking if we have previously requested to follow this user"

Reply:
I've refactored the logic based on the (0,1,2) cases that you suggested.